[Feature] Quorum Alien Species

Kearel opened the pull request:
For ParadoxSpace.

ParadoxSpace commented:
thanks Kearal!

PsiOmegaDelta commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -256,3 +256,35 @@
+
+	var/list/creatures = list()
+	for(var/mob/living/carbon/h in world)
+		creatures += h
```
Please populate/depopulate this list in mob/living/carbon/New() and Destroy(), if it is to be a thing.

Iamgoofball replied:
or just use the living mob list that already exists for purposes like this

Kearel commented:
Might already be such a list, if show_message can be sent to minds.

Iamgoofball commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -256,3 +256,35 @@
+	set category = "Abilities"
+
+	//pretty much C&P from remoteSay
+	if(usr.stat!=CONSCIOUS)
```
copypasta's bad dude

make this a standardized proc under the name of telepathic_say() and just call that instead

Kearel replied:
A definite idea. It'll allow me to clean up the three other versions of it too.

Although to be honest, I dont think anything actually uses commune.

Iamgoofball commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -256,3 +256,35 @@
+		msg = stars(msg,max(1,100-halloss) * max(1,health)/maxHealth * (max(1,100-dist/2))/100)
+		target.show_message("You hear something whisper into your ear: <i>[msg]</i>")
+		usr.show_message("You transmit: <i>[msg]</i> to [target]")
+		for(var/mob/dead/observer/G in world)
```
>looping through every single object in the world

why not loop through the dead mobs list
plus i'm like 99% certain there is a dedicated proc for showing text to all ghosts

Kearel replied:
You're right. But there isn't a dead mobs list.

GinjaNinja32 replied:
...so what's `dead_mob_list`, then?

Kearel replied:
I stand corrected. Thanks.

Iamgoofball commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -256,3 +256,35 @@
+		log_say("Quorum telepathy: [key_name(src)]-> [target.key] : [msg]")
+		msg = stars(msg,max(1,100-halloss) * max(1,health)/maxHealth * (max(1,100-dist/2))/100)
+		target.show_message("You hear something whisper into your ear: <i>[msg]</i>")
+		usr.show_message("You transmit: <i>[msg]</i> to [target]")
```
usr is not safe to use, byond has a bad habit of fucking up what usr is sometimes

Kearel replied:
Good to know.

Iamgoofball commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -256,3 +256,35 @@
+	var/dist = max(1,get_dist(usr,target)-11) //10 turf before distance effects readability
+	if(msg)
+		log_say("Quorum telepathy: [key_name(src)]-> [target.key] : [msg]")
+		msg = stars(msg,max(1,100-halloss) * max(1,health)/maxHealth * (max(1,100-dist/2))/100)
```
what is this math
why is halloss a factor here
this is insane what the fuck

Kearel replied:
It was a request so telepathy wouldn't become a dedicated radio. Halloss is specifically there so shit like tasers will effect the chances otherwise they are pretty much kidnap proof.

Iamgoofball commented on code/modules/organs/subtypes/quorum.dm:
```diff
@@ -0,0 +1,48 @@
+	// Coffee is really bad for you with busted kidneys.
+	// This should probably be expanded in some way, but fucked if I know
+	// what else kidneys can process in our reagent list.
+	var/datum/reagent/coffee = locate(/datum/reagent/drink/coffee) in owner.reagents.reagent_list
```
This should seriously be on the reagent datum's onmoblife() proc. NOT in the middle of a organ proc.

Kearel replied:
Yup. I'll see about that. On the plus side I'll do the same for original kidneys.

Zuhayr commented on code/modules/organs/subtypes/quorum.dm:
```diff
@@ -0,0 +1,48 @@
+	organ_tag = "kidneys"
+	parent_organ = "groin"
+
+/obj/item/organ/liver/quorum/process()
```
Very much not a fan of mass copypasta. You should move this down to organ level or add organ/filter or something, not duplicate code word for word.

Zuhayr commented:
This kind of duplicated code is why the species datum and organ system exist in the first place. :-1: to this unless it's significantly cleaned up and properly integrated into the existing systems.

Kearel commented:
Alright. It properly uses the organ system. Moved the coffee damage to reagents and deleted it from kidneys entirely.

Unsure how to go about telepathy_say. I have to know the target and the message, at least for the quorum dealio. Could always return the target, I guess, and let the abilities handle distribution (would also allow for their unique flavortexts regarding the messages) but I'd like to see what else needs to be done here before I get to that.

RemieRichards commented on code/__defines/species_languages.dm:
```diff
@@ -34,6 +34,7 @@
 #define LANGUAGE_ROOTSPEAK "Rootspeak"
 #define LANGUAGE_TRADEBAND "Tradeband"
 #define LANGUAGE_GUTTER "Gutter"
+#define LANGUAGE_quorum "Quorom"
```
#defines are standardised in all caps.

Kearel replied:
good catch

GinjaNinja32 replied:
also this is typoed as "Quorom", and should (AFAIK) be the same as the name of the language

Kearel replied:
I thought I caught all those... thanks.

Fox-McCloud commented:
If you're going to stereotypical bigheaded alien, I recommend VG's "greys" for sprites over this--those sprites look pretty bad, IMO.

Kearel commented:
At one point we did have greys but from what I understand one of the admins in charge of species adding doesn't like greys so it was changed to be modified humans.

I'd ask Paradox. He knows the deets, I just coded it.

Zuhayr commented:
Going to be blunt, given the lackluster response on the forums and general apathy towards this from the other devs, I don't see this getting merged. A new species is both a chunk of extra code and administration overhead, and at least in my opinion should fill both a lore and mechanical niche to offer an interesting twist on regular gameplay. This species does neither. Combined with the less than stellar code and no admin approval I am inclined to close this.

Would appreciate other dev opinions on this. @PsiOmegaDelta @GinjaNinja32 @mwerezak @Miniature @Ccomp5950 @NullSnapshot 

PS. Why did you move coffee reagent effects back into reagent code instead of organ code?

Iamgoofball commented:
@Zuhayr I asked for that for a few reasons:
1. reagent stuff being there was weird as hell
2. checking for coffee in your liver every tick seems like a bad idea
3. if it's on the reagent, it'll only run the coffee liver checks and effects if there's coffee in the system in the first place. It's small, but every step counts.

GinjaNinja32 commented:
I'm in agreement with Zuhayr here, frankly the response on the forums has been "meh" at best and there are several major issues with the code, along with what Zuhayr pointed out about extra code and admin overhead.

Kearel commented:
Fair enough. I'm sure Paradox will be disappointed but it probably wasn't ready anyways.

Closing now.

ParadoxSpace commented:
Last I recall, Virgie did say he supported me on this, as well as Serena ;n;

Miniature commented:
For the record, I'm never going to be okay with telepathy as a race's gimmick.
